packages/typescript/client/src/main.ts

```diff
@@ -9,46 +9,73 @@ export type SpawnResult = {
   bearerToken?: string
 }
 
+const ALIVE_STATUSES = ['Loading', 'Starting', 'Ready']
+
 export class SessionBackend {
   private streamReader: ReadableStreamDefaultReader | null = null
-  readonly statuses: Status[] = []
+  private statusesSeen: Set<Status> = new Set()
+
+  private fetchingStatusStream: boolean = false
+
   private _isReady: boolean = false
   private _onReady: (() => void)[] = []
+  public onReadyPromise: Promise<void>
+  private _onReadyResolve!: () => void
```

Comment:
is there a convention here to which private fields get prefixed with `_`? My slight preference is to not use `_` for private fields in TypeScript since TypeScript has a native way of representing private/public, but to use it in JavaScript since it doesn't. I'm fine with either here as long as there is a consistent convention.

Reply:
My thinking is that since this library is importable as JavaScript, the `_` prefix is the only indicator in those cases that these are not part of the public API. I've only prefixed the ones that I _really_ don't want people accessing because it could break the library (mostly functions and arrays of functions). Some of the other ones I've left unprefixed as it's not a huge deal if someone reads `statusesSeen` or `fetchStatusStream`. I don't feel super strongly about this, though.